tests/tests/test_monitor_client.py

```diff
@@ -972,3 +972,102 @@ def test_dbus_bus_filter(self, monitor_commercial_setup_no_client):
         )
         assert not "${workflow.input" in mail
         logger.info("test_dbus_bus_filter: got CRITICAL alert email.")
+
+    def test_monitorclient_men_5059(self, monitor_commercial_setup_no_client):
+        """Tests the monitor client email alerting"""
+        mailbox_path = "/var/spool/mail/local"
+        wait_for_alert_interval_s = 8
+        expected_from = "noreply@mender.io"
+        service_name = "crond"
+        user_name = "bugs.bunny@acme.org"
+        devid, _, auth, mender_device = self.prepare_env(
+            monitor_commercial_setup_no_client, user_name
+        )
+        inventory = Inventory(auth)
+        logger.info("test_monitorclient_alert_email: env ready.")
+
+        logger.info(
+            "test_monitorclient_alert_email: email alert on systemd service not running scenario."
+        )
+        prepare_service_monitoring(mender_device, service_name)
+        time.sleep(2 * wait_for_alert_interval_s)
+
+        logger.info(
+            "test_monitorclient_alert_store disabling access to docker.mender.io (point to localhost in /etc/hosts)"
+        )
+        mender_device.run("sed -i.backup -e '$a127.2.0.1 docker.mender.io' /etc/hosts")
+
+        log_file = "@tail -f /tmp/mylog.log"
+        log_pattern = "session opened for user [a-z]*"
+
+        service_name = "mylog"
+        prepare_log_monitoring(
+            mender_device, service_name, log_file, log_pattern, log_pattern_expiration=4
+        )
+        time.sleep(2 * wait_for_alert_interval_s)
+
+        mender_device.run("echo 'some line 1' >> " + log_file)
+        mender_device.run("echo 'some line 2' >> " + log_file)
+        mender_device.run(
+            "echo 'a new session opened for user root now' >> " + log_file
+        )
+        mender_device.run(
+            "echo -ne 'some line 4\nsome line 5\nsome line 6\n' >> " + log_file
+        )
+        time.sleep(2 * wait_for_alert_interval_s)
+        mender_device.run(
+            "echo 'a new session opened for user root now' >> " + log_file
+        )
+        mender_device.run("echo -ne 'some line 7\nsomeline 8\n' >> " + log_file)
+        time.sleep(4 * wait_for_alert_interval_s)
```

Comment:
then we create the check with log subsystem, and we are writing the pattern we look for ("a new session opened for user root now") into the log, so after this we have the alerts in local store.